drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -568,6 +568,36 @@ impl<'a> Handler {
                 )?;
                 Ok((RspBody::Ok, None))
             }
+            ReqBody::Policy(PolicyReq::DevOrRelease) => {
+                let mut buf = [0u8; 512];
+
+                // If the SHA-256 Digest is zeros, the CMPA is unlocked
+                self.update.read_rot_page(RotPage::Cmpa, &mut buf)?;
+                let unlocked = buf[480..].iter().all(|b| *b == 0);
+
+                // Otherwise, check which root keys are enabled
+                let dev = if unlocked {
+                    true
+                } else {
+                    self.update.read_rot_page(RotPage::CfpaActive, &mut buf)?;
+                    // Look at the ROTKH_REVOKE byte
+                    let revoke = buf[24];
+                    // Check if any of the development keys (slots 2 and 3) are
```

Comment:
I think it's impossible to add new root keys with a locked CMPA, because the 4 key hashes are stored in CMPA (which is locked).

CFPA lets us switch key _state_ for each of the 4 slots, which is a forward-only transition from invalid → valid → revoked.

Reply:
Makes sense. Thanks.